packages/components/src/List/ListItem.tsx

```diff
@@ -9,7 +9,15 @@ import { Typography } from "../Typography";
 import { Markdown } from "../Markdown";
 import { Emphasis } from "../Emphasis";
 
-export interface ListItemProps {
+export interface BaseListItemProps {
+  /**
+   * The ID of the list item. This will be helpful to know the selected list
+   * items when a batch action is implemented.
+   */
+  readonly id: number | string;
```

Comment:
something I've been thinking about using more often is `React.Key` which with our current version of react-types is exactly this

in newer versions it has `bigint` as another possible value, but I don't think that's a problem as long as we don't try to use it as an index.

you know what, this is fine. if we wanna start using `React.Key` more, I wanna put a tiny bit more thought into it.